### PR TITLE
next greater

### PR DESCRIPTION
**Pull request by megha070**
Solves #398

### CODE REVIEW

File: Algorithms/Stacks/next_greater.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
+ // this code computes the next largest element of each element in an array
+ // if the next largest element does not exist print -1 for the corresponding element
+import java.util.*;
+import java.lang.*;
+import java.io.*;
+class next_greater
+ {
+	public static void main (String[] args)throws IOException
+	{
+	BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
```
Comment (sahilbansal17) [inline Algorithms/Stacks/next_greater.java]: Incorrect indentation.

File: Algorithms/Stacks/next_greater.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            int n = Integer.parseInt(br.readLine()); 
+            long arr[] = new long[n]; 
+            String line = br.readLine(); 
+            String[] strs = line.trim().split("\\s+"); 
+            for (int i = 0; i < n; i++) 
+                arr[i] = Long.parseLong(strs[i]); 
+           Stack<Integer> st=new Stack<>();
+	    int i;
+	    long[] ans=new long[n];
+	    for(i=0;i<n;i++)
```
Comment (sahilbansal17) [inline Algorithms/Stacks/next_greater.java]: Please be consistent while using spaces between assignment statements.

**Comment by megha070** [comment]
@sahilbansal17  Please review this PR!

File: Algorithms/Stacks/next_greater.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+ // if the next largest element does not exist print -1 for the corresponding element
+import java.util.*;
+import java.lang.*;
+import java.io.*;
+class next_greater
+ {
+	public static void main (String[] args)throws IOException
+	{
+	    BufferedReader br = new BufferedReader(new InputStreamReader(System.in)); 
+        int testcase = Integer.parseInt(br.readLine());
```
Comment (sahilbansal17) [inline Algorithms/Stacks/next_greater.java]: Still not indented properly.

File: Algorithms/Stacks/next_greater.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+	            st.pop();
+	            ans[t] = arr[i];
+	        }
+	        st.push(i);//push the index of current element
+	    }
+	    for(i=0;i<n;i++)
+	    {
+	        System.out.print(ans[i]+" ");
+	    }
+        System.out.println();
```
Comment (sahilbansal17) [inline Algorithms/Stacks/next_greater.java]: This and the next two lines are also not indented properly.

**Comment by megha070** [comment]
Please review it !

**Comment by sahilbansal17** [comment]
@megha070 The code isn't properly indented now either. I don't understand how are you able to do this? 😛

**Comment by megha070** [comment]
I hope it is fine now.
I am really very sorry.

**Pull request merged**